File: server/helpers/enums/label_order.py
from enum import Enum

class LabelOrderDefault(Enum):
    FORWARD = 0
    BACKWARD = 1
    START = 2
    STOP = 3
    
    @classmethod
    def from_string(cls, label_order):
        for label in LabelOrderDefault:
            if label_order == label.name:
                return label
        return None
    
    def __str__(self) -> str:
        return self.name

File: client/lib/shared/enum/label_order.dart
import 'package:client/generated/translations.g.dart';
import 'package:easy_localization/easy_localization.dart';
import 'package:freezed_annotation/freezed_annotation.dart';

enum LabelOrderEnum {
  @JsonValue("FORWARD")
  forward,
  @JsonValue("BACKWARD")
  backward,
  @JsonValue("START")
  start,
  @JsonValue("STOP")
  stop,
}

extension LabelOrderEnumExt on LabelOrderEnum {
  String get description => {
        LabelOrderEnum.forward: tr(LocaleKeys.ActionDescription_Forward),
        LabelOrderEnum.backward: tr(LocaleKeys.ActionDescription_Backward),
        LabelOrderEnum.start: tr(LocaleKeys.ActionDescription_Start),
        LabelOrderEnum.stop: tr(LocaleKeys.ActionDescription_Stop)
      }[this]!;
}
